spanner/read.go

```diff
@@ -589,50 +611,57 @@ func (d *resumableStreamDecoder) next() bool {
 	}
 }
 
-// tryRecv attempts to receive a PartialResultSet from gRPC stream.
-func (d *resumableStreamDecoder) tryRecv(retryer gax.Retryer) {
+func (d *resumableStreamDecoder) tryRecv(mt *builtinMetricsTracer, retryer gax.Retryer) {
 	var res *sppb.PartialResultSet
-	res, d.err = d.stream.Recv()
-	if d.err == nil {
-		d.q.push(res)
-		if d.state == queueingRetryable && !d.isNewResumeToken(res.ResumeToken) {
-			d.bytesBetweenResumeTokens += int32(proto.Size(res))
-		}
-		d.changeState(d.state)
-		return
-	}
-	if d.err == io.EOF {
-		d.err = nil
-		d.changeState(finished)
-		return
-	}
-	if d.replaceSessionFunc != nil && isSessionNotFoundError(d.err) && d.resumeToken == nil {
-		// A 'Session not found' error occurred before we received a resume
-		// token and a replaceSessionFunc function is defined. Try to restart
-		// the stream on a new session.
-		if err := d.replaceSessionFunc(d.ctx); err != nil {
-			d.err = err
-			d.changeState(aborted)
-			return
+	// ignore error from gaxInvokeWithRecorder, as it is already handled in
+	// the retryer and resumableStreamDecoder.
+	_ = gaxInvokeWithRecorder(d.ctx, mt, d.method, func(ctx context.Context, settings gax.CallSettings) (context.Context, error) {
```

Comment:
So we just do not need to do anything for the returned error?

Reply:
I have updated the code to record attempt latencies in stream interceptor, and yes we don;t want to surface any errors if error while recording native metrics.